resources/sql_databases.go

```diff
@@ -453,3 +541,19 @@ func fetchSqlDatabaseDbBlobAuditingPolicies(ctx context.Context, meta schema.Cli
 	}
 	return nil
 }
+
+func fetchSqlDatabaseDbThreatDetectionPolicies(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	svc := meta.(*client.Client).Services().SQL.DatabaseThreatDetectionPolicies
+	database := parent.Item.(sql.Database)
+	details, err := client.ParseResourceID(*database.ID)
+	if err != nil {
+		return err
+	}
+	serverName := parent.Parent.Get("name").(*string)
```

Comment:
why not use parent.Parent.Item.(*...) also parent.Parent.Get can return null, so its important to check that.

Reply:
Changed to use parent.Parent.Item as suggested, thanks